### ISSUE TITLE
TCH false positive to generic runtime base classes

### ISSUE THREAD

**Issue by iurisilvio** [astral-sh/ruff#7945]
My branch with a broken example: https://github.com/astral-sh/ruff/compare/main...iurisilvio:test-7945

```python
from __future__ import annotations

from pathlib import Path
from collections import Sequence


class A(Sequence[int]):
    path: Path

# This works as expected
# class A(Sequence):
#     path: Path
``` 

```
[tool.ruff.flake8-type-checking]
runtime-evaluated-base-classes = ["collections.Sequence"]
```

Running master ruff with `ruff check --select TCH file.py` give me a `TCH003`:

```console
file.py:3:21: TCH003 Move standard library import `pathlib.Path` into a type-checking block
```

It is a false positive because ruff don't understand the base class is `collections.Sequence`. Removing the generics part `[int]` works.

**Comment by iurisilvio** [comment]
I did some investigation but was unable to fix for now. The `semantic.resolve_call_path(...)` for the `Sequence` node returns `None` instead of `collections.Sequence` and this is probably the reason it doesn't work.

https://github.com/astral-sh/ruff/blob/8255e4ed6c6dbd2eefb14160ef88fc7753649866/crates/ruff_linter/src/rules/flake8_type_checking/helpers.rs#L43

**Comment by charliermarsh** [comment]
Ah yup, makes sense -- thank you!
